kstreams/poc-ddd-aggregates/src/main/java/com/github/hpgrahsl/kafka/StreamingAggregatesDDD.java

```diff
@@ -0,0 +1,100 @@
+
+        if(args.length != 2) {
+            System.err.println("usage: java -jar <package> "
+                    + StreamingAggregatesDDD.class.getName() + " <parent_topic> <children_topic>");
+            System.exit(-1);
+        }
+
+        final String parentTopic = args[0];
+        final String childrenTopic = args[1];
+
+        Properties props = new Properties();
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streaming-aggregates-ddd");
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 10*1024);
+        props.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 1000);
+        props.put(CommonClientConfigs.METADATA_MAX_AGE_CONFIG, 500);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        final Serde<DefaultId> defaultIdSerde = SerdeFactory.createDbzEventJsonPojoSerdeFor(DefaultId.class,true);
+        final Serde<Customer> customerSerde = SerdeFactory.createDbzEventJsonPojoSerdeFor(Customer.class,false);
+        final Serde<Address> addressSerde = SerdeFactory.createDbzEventJsonPojoSerdeFor(Address.class,false);
+        final Serde<LatestAddress> latestAddressSerde = SerdeFactory.createDbzEventJsonPojoSerdeFor(LatestAddress.class,false);
+        final Serde<Addresses> addressesSerde = SerdeFactory.createDbzEventJsonPojoSerdeFor(Addresses.class,false);
+        final Serde<CustomerAddressAggregate> aggregateSerde =
+                SerdeFactory.createDbzEventJsonPojoSerdeFor(CustomerAddressAggregate.class,false);
+
+        StreamsBuilder builder = new StreamsBuilder();
+
+        //1) read parent topic i.e. customers as ktable
+        KTable<DefaultId, Customer> customerTable =
+                builder.table(parentTopic, Consumed.with(defaultIdSerde,customerSerde));
+
+        //2) read children topic i.e. addresses as kstream
+        KStream<DefaultId, Address> addressStream = builder.stream(childrenTopic,
+                Consumed.with(defaultIdSerde, addressSerde));
+
+        //2a) pseudo-aggreate addresses to keep latest relationship info
+        KTable<DefaultId,LatestAddress> tempTable = addressStream
+                .groupByKey(Serialized.with(defaultIdSerde, addressSerde))
+                .aggregate(
+                        () -> new LatestAddress(),
+                        (DefaultId addressId, Address address, LatestAddress latest) -> {
+                            latest.update(address,addressId,new DefaultId(address.getCustomer_id()));
+                            return latest;
+                        },
+                        Materialized.as(childrenTopic+"_table_temp")
+                                .withKeySerde((Serde)defaultIdSerde)
+                                    .withValueSerde(latestAddressSerde)
+                );
+
+        //2b) aggregate addresses per customer id
+        KTable<DefaultId, Addresses> addressTable = tempTable.toStream()
```

Comment:
There are some warnings about type safety when looking at this in Eclipse. Any idea how those could be avoided?

Reply:
good point, I suspect something related to lambda type inference limitations? most likely explicit typing should help. I'll look into this!

Reply:
this should also be fixed now